lib/internal/util/iterable-weak-map.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const {
+  Set,
+  SetPrototypeAdd,
+  SetPrototypeDelete,
+  Symbol,
+  WeakMap,
+  WeakMapPrototypeGet,
+  WeakMapPrototypeDelete,
+  WeakMapPrototypeSet,
+} = primordials;
+
+// This class is modified from the example code in the WeakRefs specification:
+// https://github.com/tc39/proposal-weakrefs
+// Licensed under ECMA's MIT-style license, see:
+// https://github.com/tc39/ecma262/blob/master/LICENSE.md
+class IterableWeakMap {
+  #weakMap = new WeakMap();
+  #refSet = new Set();
+  #finalizationGroup = new global.FinalizationRegistry(cleanup);
+
+  set(key, value) {
+    const ref = new global.WeakRef(key);
+    WeakMapPrototypeSet(this.#weakMap, key, { value, ref });
+    SetPrototypeAdd(this.#refSet, ref);
+    this.#finalizationGroup.register(key, {
```

Comment:
```suggestion
    FinalizationGroupPrototypeRegister(this.#finalizationGroup, key, {
```

Reply:
I don't think we'll have these prototype methods available until we get `WeakRef` working with primordials, which I believe is creating the methods [here](https://github.com/nodejs/node/blob/master/lib/internal/per_context/primordials.js#L66).

Reply:
Good call! Seems like adding WeakRef and FinalizationGroup to the list (and maybe the safe variants too) would do it

Reply:
@ljharb @aduh95 I wonder if we could freeze additional primordials late? after Node/v8 are up and running, but before executing user code.